lock-api/src/test/java/com/palantir/lock/client/SharedLockTokenTest.java

```diff
@@ -0,0 +1,95 @@
+ * limitations under the License.
+ */
+
+package com.palantir.lock.client;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import org.junit.Test;
+
+import com.palantir.lock.v2.LockToken;
+
+public class SharedLockTokenTest {
+    private static final LockToken LOCK_TOKEN = LockToken.of(UUID.randomUUID());
+
+    @Test
+    public void allSharedLocksHaveCorrectUnderlyingToken() {
+        List<LockToken> tokens = SharedLockToken.share(LOCK_TOKEN, 3);
+        convertToSharedTokens(tokens).forEach(t -> assertThat(t.referencedToken()).isEqualTo(LOCK_TOKEN));
+    }
+
+    @Test
+    public void allSharedTokensShouldHaveDifferentIds() {
+        List<LockToken> tokens = SharedLockToken.share(LOCK_TOKEN, 3);
+        assertThat(tokens.stream()
+                .map(LockToken::getRequestId)
+                .collect(Collectors.toSet())
+                .size()).isEqualTo(3);
+    }
+
+    @Test
+    public void shouldReturnUnderlyingTokenAfterAllReferencesAreUnlocked() {
+        List<LockToken> tokens = SharedLockToken.share(LOCK_TOKEN, 2);
+        SharedLockToken firstToken = convertToSharedToken(tokens.get(0));
+        SharedLockToken secondToken = convertToSharedToken(tokens.get(1));
+
+        assertThat(firstToken.unlock()).isEmpty();
+        assertThat(secondToken.unlock()).contains(LOCK_TOKEN);
+    }
+
+    @Test
+    public void callingUnlockOnSameTokenMultipleTimesShouldNotUnlockUnderlyingToken() {
+        List<LockToken> tokens = SharedLockToken.share(LOCK_TOKEN, 2);
+
+        SharedLockToken firstToken = convertToSharedToken(tokens.get(0));
+
+        assertThat(firstToken.unlock()).isEmpty();
+        assertThat(firstToken.unlock()).isEmpty();
+    }
+
+    @Test
+    public void shouldWorkAsExpectedIfThereIsOnlyOneSharedToken() {
+        List<LockToken> tokens = SharedLockToken.share(LOCK_TOKEN, 1);
+
+        assertThat(tokens.size()).isEqualTo(1);
+        SharedLockToken sharedLockToken = convertToSharedToken(tokens.get(0));
+
+        assertThat(sharedLockToken.referencedToken()).isEqualTo(LOCK_TOKEN);
+        assertThat(sharedLockToken.getRequestId()).isNotEqualTo(LOCK_TOKEN.getRequestId());
+        assertThat(sharedLockToken.unlock()).contains(LOCK_TOKEN);
+    }
+
+    @Test
+    public void shouldThrowIfCreatedWithLessThanOneReference() {
+        assertThatThrownBy(() -> SharedLockToken.share(LOCK_TOKEN, 0))
+                .hasMessage("Reference count should be more than zero");
+    }
+
+    private List<SharedLockToken> convertToSharedTokens(List<LockToken> lockTokens) {
+        return lockTokens.stream()
+                .map(this::convertToSharedToken)
+                .collect(Collectors.toList());
+    }
+
+    private SharedLockToken convertToSharedToken(LockToken lockToken) {
```

Comment:
nit: `static`

Reply:
I don't think you ever need this, wherever you use this, you always have the list, and then you extract them element by element, and then you convert. Just use `convertToSharedTokens` and then extract element by element.